### PR TITLE
softmax with mask

### PR DESCRIPTION
**Pull request by corochann**

### CODE REVIEW

File: chainer_chemistry/functions/activation/softmax.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from chainer import functions
+
+
+def softmax(x, axis=1, mask=None, mask_value=1e10):
```
Comment (mottodora) [inline chainer_chemistry/functions/activation/softmax.py]: Please add docstring.

File: chainer_chemistry/functions/activation/softmax.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    if mask is None:
+        h = x
+    else:
+        if x.shape != mask.shape:
```
Comment (mottodora) [inline chainer_chemistry/functions/activation/softmax.py]: a bit strict? But we can fix later.

Comment (mottodora) [inline chainer_chemistry/functions/activation/softmax.py]: We can check the range of value. (0. <= x <= 1.)
But we can add later.

Comment (mottodora) [reply chainer_chemistry/functions/activation/softmax.py]: Sorry. I would like to suggest we can broadcast if the dimension(of input) is consistent. So please ignore this comment now.

Comment (corochann) [reply chainer_chemistry/functions/activation/softmax.py]: yeah may be we can broadcast internally but i am not sure about all use case yet.

File: chainer_chemistry/functions/activation/softmax.py (hunk 1 above)
Comment (corochann) [reply chainer_chemistry/functions/activation/softmax.py]: ok.

File: chainer_chemistry/functions/activation/softmax.py (hunk 2 above)
Comment (corochann) [reply chainer_chemistry/functions/activation/softmax.py]: let us add later.

**Pull request merged**